src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final DateConverter DATE_CONVERTOR = new DateConverter();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        LocalDate fromDate = DATE_CONVERTOR.convert(dateFrom);
+        LocalDate toDate = DATE_CONVERTOR.convert(dateTo);
+
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String dataElement : data) {
+                String[] dataItem = dataElement.split(" ");
+                final String employeeName = dataItem[1];
+                final int hoursWorked = Integer.parseInt(dataItem[2]);
+                final int ratePerHour = Integer.parseInt(dataItem[3]);
+                LocalDate dateElement = DATE_CONVERTOR.convert(dataItem[0]);
```

Comment:
let's move indices (0, 1, 2, 3) to constants because they look like magic numbers

and if it is possible to remove `final` from variables - let's do it